src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -268,10 +270,46 @@ async def setup_user_lockfile_requests(
     )
 
 
+@dataclass(frozen=True)
+class PythonSyntheticLockfileTargetsRequest(SyntheticTargetsRequest):
+    path: str = SyntheticTargetsRequest.SINGLE_REQUEST_FOR_ALL_TARGETS
+
+
+@rule
+async def python_lockfile_synthetic_targets(
+    request: PythonSyntheticLockfileTargetsRequest,
+    python_setup: PythonSetup,
+) -> SyntheticAddressMaps:
+    if (
+        not python_setup.enable_resolves
+        or request.path != SyntheticTargetsRequest.SINGLE_REQUEST_FOR_ALL_TARGETS
+    ):
+        return SyntheticAddressMaps()
```

Comment:
It seems like implementations could avoid being called for every path if they set a ClassVar enum which declared them to be either "global" or "per-path", and then the caller called them the appropriate number of times? Or it could be a separate (but very similar) `@union` for type safety: `SyntheticGlobalTargetsRequest` vs `SyntheticDirectoryTargetsRequest`.

Reply:
Yeah, I was thinking about having two distinct union types.. maybe that is preferred, now it's the `path` field, and looking at the default value (as defined on the class) it works essentially the same as a class var enum.

Reply:
Ah, that `or request.path != ....` is actually not needed. Will remove for clarity.